nav2_util/include/nav2_util/service_client.hpp

```diff
@@ -10,32 +10,57 @@
 // distributed under the License is distributed on an "AS IS" BASIS,
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
-// limitations under the License. Reserved.
+// limitations under the License.
 
 #ifndef NAV2_UTIL__SERVICE_CLIENT_HPP_
 #define NAV2_UTIL__SERVICE_CLIENT_HPP_
 
 #include <string>
-#include "rclcpp/rclcpp.hpp"
+
 #include "nav2_util/node_utils.hpp"
+#include "rclcpp/rclcpp.hpp"
 
 namespace nav2_util
 {
 
+template<typename FutureT, typename WaitTimeT>
+std::future_status
+wait_for_result(
```

Comment:
I'm not sure I understand why this function is necessary at all. It seems like the future.wait_for(time_to_wait) would be equivalent. It seems to be doing several 100 ms waits instead of one wait. Why is that needed?

Reply:
Looking at it further my guess is that it is so that you can shutdown even if you're waiting for a response from a service call?

Reply:
I commented that out and recompiled and found that no one is using it. So, since we're not using it,  I've removed it from the the header.